PhysicsTools/Heppy/python/analyzers/objects/LeptonAnalyzer.py

```diff
@@ -187,9 +194,24 @@ def makeLeptons(self, event):
             if self.miniIsolationVetoLeptons == "inclusive":
                 for lep in event.inclusiveLeptons: 
                     self.IsolationComputer.addVetos(lep.physObj)
+            event.LeptonTrackMuPairs = []
             for lep in event.inclusiveLeptons:
                 self.attachMiniIsolation(lep)
-        
+                if self.doDirectionalIsolation:
+                    for cs in xrange(2,7): self.attachDirectionalIsolation(lep,0.1*cs)
+                if self.doMuonPlusSoftTrkPairs:
+                    self.pairedIsoChargedCandidatesMuHyp = []
+                    for pairedCand in self.IsolationComputer.findPairIsoTrack(lep.physObj, 999, 0.0, 4.0, 1.5, 5.0):
+                        _lep_p4 = lep.p4()
+                        _pc_p4 = pairedCand.p4()
+                        lep_p4 = TLorentzVector(_lep_p4.Px(),_lep_p4.Py(),_lep_p4.Pz(),_lep_p4.E())
+                        pc_p4 = TLorentzVector()
+                        pc_p4.SetPtEtaPhiM(_pc_p4.Pt(),_pc_p4.Eta(),_pc_p4.Phi(),0.105)
+                        self.pairedIsoChargedCandidatesMuHyp.append((pairedCand,(lep_p4+pc_p4).M()))
+                    self.pairedIsoChargedCandidatesMuHyp.sort(key = lambda x : x[1], reverse = True)
+                    event.LeptonTrackMuPairs.extend([(lep,x[0],x[1]) for x in self.pairedIsoChargedCandidatesMuHyp])
+                    event.LeptonTrackMuPairs.sort(key = lambda x : x[2], reverse = True)
```

Comment:
This feature also doesn't seem to me to be so general, it would better be kept in another analyzer